src/tests/test_viewsGenerationSystem.py

```diff
@@ -119,56 +78,90 @@ def tearDown(self):
+		return transformProcess
 
 	def test_transform_empty(self):
 		"""Confirms an empty transformation is successful
 		"""
-		self._test_transform()
+		self.runTransformation()
 
 	def test_transform_successfully(self):
-		"""Confirms an transformation of a single addon is successful
+		"""Confirms a transformation of a single addon runs successfully
 		"""
-		_TestDataGenerator.generate_addon()
-		self._test_transform()
+		availableApiVersions = [v.apiVer for v in _TestDataGenerator._test_nvdaAPIVersions]
+		addon = MockAddon()
+		addon.addonId = "exampleAddon"
+		addon.addonVersion = MajorMinorPatch(0, 1)
+		addon.minNvdaAPIVersion = min(availableApiVersions)
+		addon.lastTestedVersion = max(availableApiVersions)
+		addon.channel = "stable"
+		_TestDataGenerator.write_mock_addon_to_input_dir(addon)
+
+		self.runTransformation()
 
 	def test_throw_error_on_nonempty_output_folder(self):
 		"""Confirms using an existing output directory throws an error
 		"""
 		# Make the folder before transform
 		Path(DATA_DIR.OUTPUT.value).mkdir(parents=True, exist_ok=True)
 		
-		with self.assertRaises(subprocess.CalledProcessError):
-			self._test_transform()
+		with self.assertRaises(subprocess.CalledProcessError) as transformError:
+			self.runTransformation()
+		doubleEscapedDir = DATA_DIR.OUTPUT.replace('\\', '\\\\')  # stderr escapes all the backslashes twice
+		self.assertIn(
+			"FileExistsError: [WinError 183] Cannot create a file when that file already exists: "
+			f"'{doubleEscapedDir}'",
+			transformError.exception.stderr.decode("utf-8")
+		)
+
+	def _assertAddonDataWritten(self, expectedPathToAddon: str, expectedAddonVersion: MajorMinorPatch):
+		"""Confirms that an addon is written to a path and has an expected version."""
+		self.assertTrue(Path(expectedPathToAddon).exists())
+		with open(expectedPathToAddon, "r") as expectedAddon:
+			addonData = json.load(expectedAddon)
+		self.assertDictEqual(addonData["addonVersionNumber"], expectedAddonVersion._asdict())
 
 	def test_output_file_structure_matches_expected(self):
-		"""Confirms that a successful transform of multiple addons
-		is written as expected.
-		"""
-		_TestDataGenerator.generate_addon()
-		_TestDataGenerator.generate_addon_with_versions()
-		self._test_transform()
-		self._check_expected_addons_added()
+		"""Confirms that a successful transform of multiple addons is written as expected."""
+		# older version
+		addon = MockAddon()
+		addon.addonId = "multipleVersionsAddon"
+		addon.channel = "stable"
+		addon.addonVersion = MajorMinorPatch(2, 1)
+		addon.minNvdaAPIVersion = MajorMinorPatch(2020, 1)
+		addon.lastTestedVersion = MajorMinorPatch(2020, 2)
+		_TestDataGenerator.write_mock_addon_to_input_dir(addon)
+
+		# newer version
+		addon = deepcopy(addon)
+		addon.addonVersion = MajorMinorPatch(13, 0)
+		addon.minNvdaAPIVersion = MajorMinorPatch(2020, 2)
+		addon.lastTestedVersion = MajorMinorPatch(2021, 1)
+		_TestDataGenerator.write_mock_addon_to_input_dir(addon)
```

Comment:
It would be nice if this were more concise. Make it easier to comprehend even if there are more variations on the input data, get more coverage.